### PR TITLE
Add relationship reflection option

### PR DESCRIPTION
**Pull request by lgebhardt**
Addresses #727

### CODE REVIEW

**Comment by penman** [comment]
Works perfectly. Thanks @lgebhardt!

File: README.md [hunk 1]
```diff
@@ -1928,6 +1938,11 @@ JSONAPI.configure do |config|
   # Controls the serialization of resource linkage for non compound documents
   # NOTE: always_include_to_many_linkage_data is not currently implemented
   config.always_include_to_one_linkage_data = false
+
+  # Relationship reflection invokes the related resource when updates
+  # are made to a has_many relationship. By default relationship_reflection
+  # is turned off because it imposes a small performance penalty.
+  self.use_relationship_reflection = false
```
Comment (shunwen) [inline README.md]: Should be `config.use_relationship_reflection = false` ?

Comment (lgebhardt) [inline README.md]: Are you asking if the default should be to keep `use_relationship_reflection` set to true?

File: README.md [hunk 2]
```diff
@@ -1944,6 +1954,11 @@ JSONAPI.configure do |config|
   # NOTE: always_include_to_many_linkage_data is not currently implemented
   config.always_include_to_one_linkage_data = false
 
+  # Relationship reflection invokes the related resource when updates
+  # are made to a has_many relationship. By default relationship_reflection
+  # is turned off because it imposes a small performance penalty.
+  self.use_relationship_reflection = false
```
Comment (shunwen) [inline README.md]: Sorry, I mean `self` should be `config`.

Comment (lgebhardt) [inline README.md]: @shunwen Good catch, thanks!

**Pull request merged**